src/activity_handlers.cpp

```diff
@@ -1784,6 +1785,20 @@ void activity_handlers::gunmod_add_finish( player_activity *act, player *p )
     }
 }
 
+void activity_handlers::toolmod_add_finish( player_activity *act, player *p )
+{
+    act->set_to_null();
+    if( act->values.size() != 1 ) {
+        debugmsg( "Insufficient arguments to ACT_TOOLMOD_ADD" );
+        return;
+    }
+    item &tool = p->i_at( act->position );
+    item &mod = p->i_at( act->values[0] );
```

Comment:
No, they'll be modified: `tools.contents` will have a new element, and `mod` will be removed.